Assets/Plugins/Editor/Uplift/DependencyResolution/Resolver.cs

```diff
@@ -0,0 +1,255 @@
+// --- BEGIN LICENSE BLOCK ---
+/*
+ * Copyright (c) 2019-present WeWantToKnow AS
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+// --- END LICENSE BLOCK ---
+
+
+using Uplift.Common;
+using Uplift.Packages;
+using Uplift.Schemas;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Text;
+
+namespace Uplift.DependencyResolution
+{
+	class Resolver : IDependencySolver
+	{
+		public PackageList packageList;
+		DependencyGraph baseGraph = new DependencyGraph();
+		LinkedList<DependencyDefinition> originalDependencies;
+		Stack<State> stateStack = new Stack<State>();
+
+		public Resolver(DependencyGraph baseGraph, PackageList packageList)
+		{
+
+			this.baseGraph = baseGraph;
+			this.packageList = packageList;
+			if (packageList.GetType() == typeof(PackageListStub))
```

Comment:
I don't like this at.

We don't want to have the Resolver know whether the PackageList is a stub or not, we want the classes to offer the same interface and behave differently